File: c/bit32.c
// bit32.c
//
// A drop-in replacement for Lua's bit32 library,
// designed for Noahvim.
//
// This file does not contain the necessary boilerplate
// for use with Lua's C API. Instead, use this library
// with LuaJIT's FFI library.
//
// Licensed under the MIT License (see LICENSE).
//
// Noah Trupin
// September 14, 2023

int bor(int lhs, int rhs) {
    return lhs | rhs;
}

int band(int lhs, int rhs) {
    return lhs & rhs;
}

int rshift(int lhs, int rhs) {
    return lhs >> rhs;
}

File: lua/ffi/bit32.lua
-- bit32.lua
-- ffi for bit32.c

local ffi = require("ffi")

ffi.cdef [[
  int bor(int, int);
  int band(int, int);
  int rshift(int, int);
]]

return ffi.load(os.getenv("HOME") .. "/.config/nvim/build/bit32.so")
